sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -310,30 +310,31 @@ def text(self, encoding: Optional[str] = None) -> str:
         # implement the decoding explicitly here
         body = self.body()
 
-        ctype = self.headers.get(aiohttp.hdrs.CONTENT_TYPE, "").lower()
-        mimetype = aiohttp.helpers.parse_mimetype(ctype)
-
-        encoding = mimetype.parameters.get("charset")
-        if encoding:
-            try:
-                codecs.lookup(encoding)
-            except LookupError:
-                encoding = None
         if not encoding:
-            if mimetype.type == "application" and (
-                    mimetype.subtype == "json" or mimetype.subtype == "rdap"
-            ):
-                # RFC 7159 states that the default encoding is UTF-8.
-                # RFC 7483 defines application/rdap+json
-                encoding = "utf-8"
-            elif body is None:
-                raise RuntimeError(
-                    "Cannot guess the encoding of a not yet read body"
-                )
-            else:
-                encoding = chardet.detect(body)["encoding"]
-        if not encoding:
-            encoding = "utf-8-sig"
+            ctype = self.headers.get(aiohttp.hdrs.CONTENT_TYPE, "").lower()
+            mimetype = aiohttp.helpers.parse_mimetype(ctype)
+
+            encoding = mimetype.parameters.get("charset")
+            if encoding:
```

Comment:
Why you put if encoding in if not encoding: statement?

Reply:
This is original implementation's L.317.
On original L.316, the method overwrites `encoding` if it explicitly specified by callers.
I thought that original L.313 to L.336 is the guard for `encoding = None` so just add L.313 `if not encoding:` and indented following lines.

But reviewing my own code, this `None` check might be enough just before original L.316.
So I re-write patch and force-pushed it.

Thanks to your review!